docs/components/index.md

```diff
@@ -0,0 +1,315 @@
+
+* The `HeadingFontStyle` property to the CSS property value for `font-style`.
+* `HeadingText` to the content of the `<h1>` element.
+
+```cshtml
+<h1 style="font-style:@HeadingFontStyle">@HeadingText</h1>
+```
+
+### Two-way databinding
+
+Two-way databinding to both components and DOM elements is accomplished with the `bind` attribute. The following example binds the `ItalicsCheck` property to the checkbox's `checked` attribute:
+
+```cshtml
+<input type="checkbox" class="form-check-input" 
+    id="italicsCheck" bind="@ItalicsCheck">
+```
+
+Selecting the checkbox in the UI updates the property's value. Updating the property value in C# code updates the checkbox in the UI.
+
+**Format strings**
+
+Two-way databinding works with DateTime format stings (but not other format expressions at this time, such as currency or number formats):
+
+```cshtml
+<input type="date" bind="@StartDate" format="yyyy-MM-dd">
+
+@functions {
+    public DateTime StartDate { get; set; } = 
+        new DateTime(2020, 1, 1);
+}
+```
+
+**Component attributes**
+
+Binding also recognizes component attributes, where `bind-{property}` can bind a property value across components.
+
+The following parent component uses `ChildComponent` and binds the value `1979` from `ParentYear` to the child component's `Year` property:
+
+Parent component:
+
+```cshtml
+<ChildComponent bind-Year="@ParentYear" />
+
+@functions {
+    public int ParentYear { get; set; } = 1979;
+}
+```
+
+Child component:
+
+```cshtml
+<div> ... </div>
+
+@functions {
+    public int Year { get; set; }
+    public Action<int> ValueChanged { get; set; }
+}
+```
+
+**User-defined binding**
+
+Other mappings for `bind` can be created:
+
+```cshtml
+[BindElement("ul", "foo", "myvalue", "myevent")]
+public class BindAttributes
+{
+}
+```
+
+*MyComponent.cshtml*:
+
+```cshtml
+<ul bind-foo="@SomeExpression" />
+```
+
+**Bind value and change handler attributes**
+
+`bind` can be specified with any value attribute name and change handler attribute name. The following example binds the current value of `SomeExpression` to `myvalue` and a change handler lambda to `myevent`:
+
+```cshtml
+<ul bind-myvalue-myevent="@SomeExpression">
+    ...
+</ul>
+```
+
+## Event handling
+
+Blazor offers two event handling features, `@onclick` and `@onchange`. The following code calls the `UpdateHeading` method when the button is selected in the UI:
+
+```cshtml
+<button class="btn btn-primary" @onclick(UpdateHeading)>
+    Update heading
+</button>
+```
+
+The following code calls the `CheckboxChanged` method when the checkbox is changed in the UI:
+
+```cshtml
+<input type="checkbox" class="form-check-input" 
+    id="callsMethodCheck" @onchange(CheckboxChanged)>
+```
+
+`OnInit` and `OnInitAsync` execute code after the component has been initialized:
+
+```csharp
+protected override void OnInit()
+{
+    ...
+}
+```
+
+```csharp
+protected override async Task OnInitAsync()
```

Comment:
Could we specify why the two both exist? (i.e., if you want to perform an asynchronous operation, use `OnInitAsync` and use the `await` keyword in your code, otherwise simply use `OnInit`).